### DISCUSSION TITLE
Connection limit

### DISCUSSION THREAD

**Discussion by jorendorff** [tokio-rs/axum#2561] [Q&A] [answered]
### Summary

Is there a way to cap the number of connections (in the sense of TCP connections, or socket fds) that `axum::serve` will accept?

I'm already using a tower ConcurrencyLimit, but I believe it only limits the number of in-flight requests. HTTP connections can also be idle, between requests, in which case I don't think they'd count toward that limit.

### axum version

0.7.4

**Comment by davidpdrsn** [comment, marked as answer]
Not currently. I'd recommend you adapt https://github.com/tokio-rs/axum/blob/main/examples/serve-with-hyper/src/main.rs.
